packages/@aws-cdk/aws-codepipeline-api/lib/test-action.ts

```diff
@@ -0,0 +1,42 @@
+import cdk = require("@aws-cdk/cdk");
+import { Action, ActionCategory, CommonActionProps } from "./action";
+import { Artifact } from "./artifact";
+
+/**
+ * Construction properties of the low-level {@link TestAction test Action}.
+ */
+export interface TestActionProps extends CommonActionProps {
+  /**
+   * The source to use as input for this test.
+   */
+  inputArtifact: Artifact;
+
+  /**
+   * The service provider that the action calls.
+   *
+   * @example 'CodeBuild'
+   */
+  provider: string;
+
+  /**
+   * The action's configuration. These are key-value pairs that specify input values for an action.
+   * For more information, see the AWS CodePipeline User Guide.
+   *
+   * http://docs.aws.amazon.com/codepipeline/latest/userguide/reference-pipeline-structure.html#action-requirements
+   */
+  configuration?: any;
+}
+
+export abstract class TestAction extends Action {
```

Comment:
Add some documentation here explaining that this is an abstract class and reference the concrete implementations and/or APIs such as `addTestToPipeline`

Reply:
Done.